src/main/kotlin/com/terraformation/backend/documentproducer/db/VariableStore.kt

```diff
@@ -132,7 +132,7 @@ class VariableStore(
                   .where(replacementVariables.REPLACES_VARIABLE_ID.eq(ID)))
           .groupBy(STABLE_ID)
           .fetch()
-          .map { fetchVariable(it[DSL.max(ID)]!!) }
+          .mapNotNull { fetchVariable(it[DSL.max(ID)]!!) }
           .sortedBy { it.deliverablePosition }
```

Comment:
I am surprised that Kotlin didn't throw an error here previously, seeing that `it` would be a `Variable?` here.

Reply:
This puzzled me too. Turns out it's because the `map()` being called here isn't the Kotlin one, but rather a method on the jOOQ `Result` class, and since it's a Java function without an explicit nullability annotation, the return value is `List<Variable!>` rather than `List<Variable?>` which disables the nullability check on the next line.